src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -182,6 +237,25 @@ public static void disableCache() {
     return resolveRawArguments(resolveGenericType(type, subType), subType);
   }
 
+  /**
+   * The same as {@link #substituteGenerics(Type, Class)}, but first resolves the generic type of {@code type}.
+   * This is a convenience method.
```

Comment:
We should document a `@throws UnsupportedOperationException` that describes what is unsupported

Reply:
Well, this method is just pure convenience for consuming the API, I would like to avoid documenting it with anything more than just stating this and point the reader at the "real" method.

Reply:
Ah, sure. I meant that both of them should be documented.